esp/src/src-react/components/FileParts.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from "react";
+import { useConst } from "@fluentui/react-hooks";
+import * as Observable from "dojo/store/Observable";
+import { Memory } from "src/Memory";
+import nlsHPCC from "src/nlsHPCC";
+import { useFile } from "../hooks/File";
+import { HolyGrail } from "../layouts/HolyGrail";
+import { DojoGrid } from "./DojoGrid";
+
+interface FilePartsProps {
+    cluster?: string;
+    logicalFile: string;
+}
+
+export const FileParts: React.FunctionComponent<FilePartsProps> = ({
+    cluster,
+    logicalFile
+}) => {
+
+    const [file, , _refresh] = useFile(cluster, logicalFile);
+    const [grid, setGrid] = React.useState<any>(undefined);
+    const [, setSelection] = React.useState([]);
+
+    //  Grid ---
+    const gridStore = useConst(new Observable(new Memory("Id")));
+    const gridSort = useConst([{ attribute: "Id", "descending": false }]);
+    const gridQuery = useConst({});
+    const gridColumns = useConst({
+        Id: { label: nlsHPCC.Part, sortable: true, },
+        Copy: { label: nlsHPCC.Copy, sortable: true, },
+        Ip: { label: nlsHPCC.IP, sortable: true, },
+        Cluster: { label: nlsHPCC.Cluster, sortable: true, },
+        Partsize: { label: nlsHPCC.Size, sortable: true, },
+        // CompressedSize: { label: nlsHPCC.CompressedSize, sortable: true, },
+    });
+
+    const refreshTable = (clearSelection = false) => {
+        grid?.set("query", gridQuery);
+        if (clearSelection) {
+            grid?.clearSelection();
+        }
+    };
+
+    React.useEffect(() => {
+        if (file?.DFUFilePartsOnClusters) {
+            const fileParts = file?.DFUFilePartsOnClusters?.DFUFilePartsOnCluster[0]?.DFUFileParts?.DFUPart;
+            gridStore.setData(fileParts.map(part => {
+                return {
+                    Id: part.Id,
+                    Copy: part.Copy,
+                    Ip: part.Ip,
+                    Cluster: cluster,
+                    Partsize: part.Partsize,
+                    // CompressedSize: part.Partsize
```

Comment:
Should be included, but will need a formatter - see d3Format in https://github.com/hpcc-systems/HPCC-Platform/blob/bf8277cc93b1cb86cfc5fadf63dc33bab6260965/esp/src/src/WUScopeController.ts